sklearn/linear_model/coordinate_descent.py

```diff
@@ -372,16 +372,26 @@ def enet_path(X, y, l1_ratio=0.5, eps=1e-3, n_alphas=100, alphas=None,
     ElasticNet
     ElasticNetCV
     """
-    # We expect X and y to be already float64 Fortran ordered when bypassing
+    # We expect X and y to be already Fortran ordered when bypassing
     # checks
     if check_input:
-        X = check_array(X, 'csc', dtype=np.float64, order='F', copy=copy_X)
-        y = check_array(y, 'csc', dtype=np.float64, order='F', copy=False,
+        if X.dtype is np.float32:
+            X = check_array(X, 'csc', dtype=np.float32, order='F', copy=copy_X)
+            y = check_array(y, 'csc', dtype=np.float32, order='F', copy=False,
                         ensure_2d=False)
-        if Xy is not None:
-            # Xy should be a 1d contiguous array or a 2D C ordered array
-            Xy = check_array(Xy, dtype=np.float64, order='C', copy=False,
-                             ensure_2d=False)
+            if Xy is not None:
+                # Xy should be a 1d contiguous array or a 2D C ordered array
+                Xy = check_array(Xy, dtype=np.float32, order='C', copy=False,
+                                 ensure_2d=False)
+        else:
+            X = check_array(X, 'csc', dtype=np.float64, order='F', copy=copy_X)
+            y = check_array(y, 'csc', dtype=np.float64, order='F', copy=False,
+                            ensure_2d=False)
+            if Xy is not None:
+                # Xy should be a 1d contiguous array or a 2D C ordered array
+                Xy = check_array(Xy, dtype=np.float64, order='C', copy=False,
+                                 ensure_2d=False)
```

Comment:
please address this! and yet doesn't `check_array` allow for a list of dtypes in order of preference, so that `dtype=[np.float64, np.float32]` should do the trick?